scripts/poms/sbndpoms_caf_concatenation.sh

```diff
@@ -0,0 +1,290 @@
+#!/bin/bash
+
+# Define the output directories
+# These will be appended with metadata info
+WORKDIR="/pnfs/sbnd/scratch/sbndpro/mcp"
+OUTDIR="/pnfs/sbnd/persistent/sbndpro/mcp"
+# Concatenate files together up to 1GB
+MAXSIZE=1000000000
+
+while :; do
+  case $1 in
+    -h|-\?|--help)
+      show_help    # Display a usage synopsis.
+      exit
+      ;;
+    --defname)       # Takes an option argument; ensure it has been specified.
+      if [ "$2" ]
+      then
+        DEFNAME="$2"
+        shift
+      else
+        echo "$0 ERROR: defname requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --workdir)       # Takes an option argument; ensure it has been specified.
+      if [ "$2" ]
+      then
+        WORKDIR="$2"
+        shift
+      else
+        echo "$0 ERROR: workdir requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --outdir)       # Takes an option argument; ensure it has been specified.
+      if [ "$2" ]
+      then
+        OUTDIR="$2"
+        shift
+      else
+        echo "$0 ERROR: outdir requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --size)       # Takes an option argument; ensure it has been specified.
+      if [ "$2" ]
+      then
+        MAXSIZE="$2"
+        shift
+      else
+        echo "$0 ERROR: size requires a non-empty option argument."
+        exit 1
+      fi
+      ;;
+    --flatten)       # Takes an option argument; ensure it has been specified.
+      FLATTEN=true
+      shift
+      ;;
+    *) break
+  esac
+  shift
+done
+
+if  [ -z "$DEFNAME" ]
+then
+  echo "$0 ERROR: defname is mandatory"
+  exit 2
+fi
+
+if  [ -z "$WORKDIR" ]
+then
+  echo "$0 ERROR: workdir is mandatory"
+  exit 2
+fi
+
+if  [ -z "$OUTDIR" ]
+then
+  echo "$0 ERROR: outdir is mandatory"
+  exit 2
+fi
+
+if  [ -z "$MAXSIZE" ]
+then
+  echo "$0 ERROR: size is mandatory"
+  exit 2
+fi
+
+echo "Running with settings:"
+echo "  Definition: $DEFNAME"
+echo "  Workdir: $WORKDIR"
+echo "  Outdir: $OUTDIR"
+echo "  Max Size: $(numfmt --to=si $MAXSIZE)"
+if [ ! -z "$FLATTEN" ]
+then
+  echo "  Making Flat CAFS"
+fi
+
+prepare()
+{
+  # Assume that the metadata is roughly the same in all files, so take the first
+  FILENAME=$(samweb -e sbnd list-definition-files $DEFNAME | head -n 1)
+  samweb -e sbnd get-metadata $FILENAME --json > base.json
+
+  # Lets change "caf" to "concat_caf" for the stage and defname
+  MDPRODUCTIONDEFNAME=$(echo "$DEFNAME" | sed "s/caf/concat_caf/g")
+
+  # Now we want to set some global vars to define the ouput path
+  MDFILETYPE=$(cat base.json | grep "file_type" | cut -d\" -f4)
```

Comment:
Nitpick: these can all be `grep base.json "something"`
Potentially `jq` is a better way to deal with json on the command line.
Or take a look at `/cvmfs/nova.opensciencegrid.org/externals/novaproduction/v05.22/NULL/bin/get_metadata_field` (it's a python script). Maybe we want something like that?